payjoin/src/send/mod.rs

```diff
@@ -616,20 +624,342 @@ mod test {
     }
 
     #[test]
-    fn test_receiver_steals_sender_change() -> Result<(), BoxError> {
+    fn test_disable_output_substitution_query_param() -> Result<(), BoxError> {
+        let url = serialize_url(
+            Url::parse("http://localhost")?,
+            OutputSubstitution::Disabled,
+            None,
+            FeeRate::ZERO,
+            "2",
+        );
+        assert_eq!(url, Url::parse("http://localhost?v=2&disableoutputsubstitution=true")?);
+
+        let url = serialize_url(
+            Url::parse("http://localhost")?,
+            OutputSubstitution::Enabled,
+            None,
+            FeeRate::ZERO,
+            "2",
+        );
+        assert_eq!(url, Url::parse("http://localhost?v=2")?);
+        Ok(())
+    }
+
+    // ***************** Tests for Sender's payjoin proposal checklist from BIP-78 *****************
+    #[test]
+    fn test_transaction_versions_dont_match() -> Result<(), BoxError> {
         let ctx = create_psbt_context()?;
-        let mut proposal = PARSED_PAYJOIN_PROPOSAL_WITH_SENDER_INFO.clone();
-        for output in proposal.outputs_mut() {
-            output.bip32_derivation.clear();
+        let mut proposal: bitcoin::Psbt = PARSED_PAYJOIN_PROPOSAL.clone();
+
+        let original_version = ctx.original_psbt.unsigned_tx.version;
+        let proposed_version = Version::non_standard(88);
+        proposal.unsigned_tx.version = proposed_version;
+
+        let result = ctx.process_proposal(proposal);
+        assert!(
+            result.is_err(),
+            "Process proposal should reject proposal since the version is different than the original psbt"
+        );
+
+        match result {
+            Ok(_) => panic!("Expected error, got success"),
```

Comment:
Isn't this assert/panic pattern redundant? I think this is effectively the same thing as:

```suggestion
        match result {
            Ok(_) => panic!("Process proposal should reject proposal since the version is different than the original psbt"),
```

Reply:
I think so. There was an existing test for one of the sender checks in this format - I kept the new tests in the same format but I'll update these to remove the redundancy